axis.go

```diff
@@ -497,6 +497,129 @@ func (utt UnixTimeTicks) Ticks(min, max float64) []Tick {
 	return ticks
 }
 
+// TimeTicks is suitable for axes representing time values.
+// TimeTicks expects values in Unix time seconds.  It will
+// adjust the number of ticks according to the specified Width.  If
+// not specified, Width defaults to 10 centimeters.
+type TimeTicks struct {
+	// Width is the width of the underlying graph, used to calculate
+	// the number of ticks that can fit properly with their time
+	// shown.
+	Width vg.Length
+}
+
+var _ Ticker = TimeTicks{}
+
+// Inspired by https://github.com/d3/d3-scale/blob/master/src/time.js
+var tickRules = []tickRule{
+	{time.Millisecond, "15:04:05.000", "15:04:05", ".000"},
+	{200 * time.Millisecond, "15:04:05.000", "15:04:05", ".000"},
+	{500 * time.Millisecond, "15:04:05.000", "15:04:05", ".000"},
+	{time.Second, "15:04:05", "15:04", ":05"},
+	{2 * time.Second, "15:04:05", "15:04", ":05"},
+	{5 * time.Second, "15:04:05", "15:04", ":05"},
+	{15 * time.Second, "Jan 02, 15:04", "Jan 02", "15:04"},
+	{30 * time.Second, "15:04:05", "15:04", ":05"},
+	{time.Minute, "15:04:05", "15:04", ":05"},
+	{2 * time.Minute, "Jan 02, 3:04pm", "Jan 02", "3:04pm"},
+	{5 * time.Minute, "Jan 02, 3:04pm", "Jan 02", "3:04pm"},
+	{15 * time.Minute, "Jan 02, 3:04pm", "Jan 02", "3:04pm"},
+	{30 * time.Minute, "Jan 02, 3:04pm", "Jan 02", "3:04pm"},
+	{time.Hour, "Jan 2, 3pm", "Jan 2", "3pm"},
+	{3 * time.Hour, "Jan 2, 3pm", "Jan 2", "3pm"},
+	{6 * time.Hour, "Jan 2, 3pm", "Jan 2", "3pm"},
+	{12 * time.Hour, "Jan 2", "Jan 2", "3pm"},
+	{24 * time.Hour, "Jan 2", "Jan", "2"},
+	{48 * time.Hour, "Jan 2", "Jan", "2"},
+	{7 * 24 * time.Hour, "Jan 2", "Jan", "2"},
+	{month, "Jan 2006", "2006", "Jan"},
+	{3 * month, "Jan 2006", "2006", "Jan"},
+	{6 * month, "Jan 2006", "2006", "Jan"},
+	{12 * month, "2006", "", "2006"},
+	{2 * year, "2006", "", "2006"},
+	{5 * year, "2006", "", "2006"},
+	{10 * year, "2006", "", "2006"},
+}
+
+const month = 31 * 24 * time.Hour
```

Comment:
These can be in a const block. Also, I'm sort of troubled by the choice of length of month; the marginally closer length is 30 days.